noxfile.py

```diff
@@ -8,9 +8,10 @@
 from functools import lru_cache
 from glob import glob
 from pathlib import Path
-from typing import Any, Callable, Dict, List, Optional, Tuple
+from typing import Any, Callable, Dict, Iterator, List, Optional, Tuple
```

Comment:
This is admittedly unrelated, but aren't the `typing.*` versions of things like `Dict`, `List` and `Iterator` deprecated since 3.9, i.e. shouldn't we just use `dict[..]`, `list[..]` and `collections.abc.Iterator` with subscripting instead or do we need the Nox script to work on older Python versions?

Reply:
Yeah the noxfile is used by all our supported versions for various CI steps. We could probably use `from __future__ import annotations` and tidy this now, if you like?

Reply:
No, let's leave it backwards compatible for now and not spend too much time here when we have other more pressing matters to attend to.